### PR TITLE
Replace Gizmo with CubeGizmo

### PR DESCRIPTION
**Pull request by soshochang**
CubeGizmo with cube, xyz axis, and triangle controls.

CubeGizmo orientation matches the orientation of the CAD part
Clicking on a cube face orients the camera to that face
Clicking a triangle arrow rotates the camera

20240429_CubeGizmo

### CODE REVIEW

**Comment by av8ta** [comment]
Hey @soshochang this looks great! Much better than simply fixing the broken gizmo, thanks :)

I've pulled your code down and run it and the cube gizmo is working. Just a few nits to sort out, I'll leave some comments.

File: applications/web/src/routes/(CADmium)/CubeGizmo/CubeGizmo.d.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import type { Key, ThrelteUseTaskOptions } from '@threlte/core'
+import { SvelteComponent } from 'svelte'
+import type { ColorRepresentation } from 'three'
+import type { SetCameraFocus } from "shared/types"
```
Comment (av8ta) [inline applications/web/src/routes/(CADmium)/CubeGizmo/CubeGizmo.d.ts:4]: Mixture of formatting. Please run the format script.

File: applications/web/src/routes/(CADmium)/CubeGizmo/CubeGizmo.d.ts
Comment (av8ta) [inline applications/web/src/routes/(CADmium)/CubeGizmo/CubeGizmo.d.ts:1]: I want to move away from global `*.d.ts` files, please rename them to `*.ts` without the `.d.ts`

File: applications/web/src/routes/(CADmium)/CubeGizmo/CubeGizmo.svelte.d.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { SvelteComponent } from "svelte";
+import type { CubeGizmoEvents, CubeGizmoProps, CubeGizmoSlots } from './CubeGizmo';
```
Comment (av8ta) [inline applications/web/src/routes/(CADmium)/CubeGizmo/CubeGizmo.svelte.d.ts:2]: `CubeGizmoEvents` and `CubeGizmoSlots` don't appear to be exported anymore.

**Review by av8ta — COMMENTED** [review]
@MattFerraro are you happy with me to go ahead and merge this after a few minor things are resolved?

**Comment by MattFerraro** [comment]
Looks great! Yes, after the small changes you requested are in, please go ahead and merge!

**Comment by soshochang** [comment]
Done! btw I will be working next on adding some more functionality to this

**Comment by av8ta** [comment]
Merged & live! https://mattferraro.github.io/CADmium/ Thanks again that's a great improvement :)

**Pull request merged**